src/project.py

```diff
@@ -31,6 +32,32 @@ def context(pname, output_format=None):
     formatter = formatters.get(output_format)
     return formatter(cfngen.build_context(pname, stackname=core.mk_stackname(pname, "test")))
 
+@requires_project
+def clone_project_formulas(pname):
+    """clones a project's list of formulas to `cloned-projects/$formulaname`, if it doesn't already exist.
+    if it does exist, it attempts to update it with a `git pull`."""
+    destination = config.PROJECT_FORMULAS
+    pdata = project.project_data(pname)
+
+    formula_url_list = [pdata.get('formula-repo')]
+    formula_url_list.extend(pdata.get('formula-dependencies', []))
```

Comment:
sounds like this does a lot of `git pull` in `builder-base-formula`; perhaps keep track of what has already been pulled, though don't want to make this method stateful?